File: clients/algoliasearch-client-kotlin/client/src/commonMain/kotlin/com/algolia/client/configuration/ChunkedHelperOptions.kt
package com.algolia.client.configuration

/**
 * Optional configuration for chunked helpers that batch records and poll for task completion.
 *
 * Designed to grow over time; future shared helper config (e.g. `timeout`, `batchSize` defaults)
 * should be added here instead of widening every helper's parameter list.
 *
 * @param maxRetries Maximum number of retries when polling for task completion. Defaults to 100.
 */
public data class ChunkedHelperOptions(
  public val maxRetries: Int = 100,
)

File: clients/algoliasearch-client-scala/src/main/scala/algoliasearch/extension/ChunkedHelperOptions.scala
package algoliasearch.extension

/** Optional configuration shared across chunked helpers that batch records and poll for task completion.
  *
  * Designed to grow over time; future shared helper config (e.g. batchSize, timeout) should be added here instead of
  * widening every helper's parameter list.
  *
  * @param maxRetries
  *   The maximum number of retries when polling for task completion. Defaults to
  *   [[ChunkedHelperOptions.DefaultMaxRetries]].
  */
case class ChunkedHelperOptions(
    maxRetries: Int = ChunkedHelperOptions.DefaultMaxRetries
)

object ChunkedHelperOptions {

  /** The default maximum number of retries for chunked helpers polling for task completion. */
  val DefaultMaxRetries: Int = 100

  /** The default maximum number of retries for `replaceAllObjects` polling for task completion. */
  val DefaultReplaceAllObjectsMaxRetries: Int = 800
}
